src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6SignalBuilder.java

```diff
@@ -0,0 +1,158 @@
+package frc.robot.hardware.signal.phoenix;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusSignal;
+import frc.robot.hardware.signal.AngleSignal;
+import frc.robot.hardware.signal.DoubleSignal;
+import frc.utils.AngleUnit;
+import frc.utils.ctre.PhoenixProUtils;
+
+public class Phoenix6SignalBuilder {
+
+	private static final int UPDATE_FREQUENCY_RETRIES = 5;
+
+	public interface SignalGetter {
+
+		StatusSignal<Double> getSignal();
+
+	}
```

Comment:
if you want to put all those classes in one file, then at least put them all at the bottom of the file